### PR TITLE
libs: add separate langgraph cli, sdk-py, sdk-js libraries and move core langgraph

### PR DESCRIPTION
**Pull request by vbarda**
This change introduces several new langgraph libraries and a new repository structure:
- `langgraph` and other libraries will now live under `/libs` directory
- adds `langgraph_cli` for starting/managing LangGraph API server `/libs/cli`
- adds Python SDK for interacting with LangGraph API (`libs/sdk-py`)
- adds JS SDK for interacting with LangGraph API (`libs/sdk-js`)
- update CI that runs lint and test checks across all libraries
- add a new `pyproject.toml` for managing requirements for publishing docs

### CODE REVIEW

**Comment by nfcampos** [comment]
@vbarda I made some updates to sdk today, lets make sure we dont lose those

**Comment by vbarda** [comment]
> @vbarda I made some updates to sdk today, lets make sure we dont lose those

yup, i'll update it

File: libs/sdk-js/index.d.cts [hunk 1]
```diff
@@ -0,0 +1 @@
+export * from './dist/index.cjs'
```
Comment (hinthornw) [inline libs/sdk-js/index.d.cts:1]: noob question - aren't these generated by `yarn build`? Or are we intending to check them in?

Comment (vbarda) [reply libs/sdk-js/index.d.cts:1]: actually great catch, i accidentally checked them in, let me remove

**Review by nfcampos — APPROVED** [review]
lgtm pending adding root licence

File: pyproject.toml [hunk 2]
```diff
@@ -1,84 +1,26 @@
 [tool.poetry]
-name = "langgraph"
-version = "0.0.69"
-description = "langgraph"
+name = "langgraph-monorepo"
```
Comment (hinthornw) [inline pyproject.toml:2]:
```
Warning: The current project could not be installed: No file/folder found for package langgraph-monorepo
...

If you want to use Poetry only for dependency management but not for packaging, you can disable package mode by setting package-mode = false in your pyproject.toml file.
In a future version of Poetry this warning will become an error!
```

Comment (vbarda) [reply pyproject.toml:2]: nice, fixed!

Comment (vbarda) [reply pyproject.toml:2]: ugh, unfortunately it doesn't work w/ older versions of poetry, will make a TODO to bump poetry version in our CI and then fix up

**Pull request merged**